packages/tpetra/core/src/Tpetra_FECrsGraph_decl.hpp

```diff
@@ -465,10 +465,10 @@ namespace Tpetra {
     //@{
 
     //! Migrates data to the owned mode
-    void endFill();
+    void endAssembly();
 
-    //! Activates the owned+shared mode for assembly.  This can only be called once.
-    void beginFill();
+    //! Activates the owned+shared mode for assembly
+    void beginAssembly();
```

Comment:
We need an endAssembly(domainMap, rangeMap), like the fillComplete overload; and fillComplete should be private.  This is particularly needed for rectangular matrices.

Reply:
You got it!